lib/tables.c

```diff
@@ -482,6 +545,75 @@ node_table_dump_text(node_table_t *self, FILE *out)
     return ret;
 }
 
+int
+node_table_load_text(node_table_t *node_table, FILE *file)
+{
+    int ret = MSP_ERR_FILE_FORMAT;
+    int err;
+    size_t k;
+    size_t MAX_LINE = 1024;
+    char *line = NULL;
+    double time;
+    int flags, population, individual, id, is_sample;
+    char *name;
+    const char *header = "id\tis_sample\ttime\tpopulation\tindividual\tmetadata\n";
+    char *start;
+
+    line = malloc(MAX_LINE);
+    k = MAX_LINE;
+
+    ret = node_table_clear(node_table);
+    if (ret < 0) {
+        goto out;
+    }
+
+    // check the header
+    err = getline(&line, &k, file);
+    if (err < 0) {
+        goto out;
+    }
+    err = strcmp(line, header);
+    if (err != 0) {
+        goto out;
+    }
+
+    while ((err = getline(&line, &k, file)) != -1) {
+        start = line;
+        err = get_sep_atoi(&start, &id, '\t');
+        if (err <= 0) {
+            goto out;
+        }
+        err = get_sep_atoi(&start, &is_sample, '\t');
+        flags = (is_sample && MSP_NODE_IS_SAMPLE);
+        if (err <= 0) {
+            goto out;
+        }
+        err = get_sep_atof(&start, &time, '\t');
+        if (err <= 0) {
+            goto out;
+        }
+        err = get_sep_atoi(&start, &population, '\t');
+        if (err <= 0) {
+            goto out;
+        }
+        err = get_sep_atoi(&start, &individual, '\t');
+        if (err <= 0) {
+            goto out;
+        }
+        err = get_sep_atoa(&start, &name, '\n');
+        if (err < 0 || *start != '\0') {
+            goto out;
+        }
+        ret = node_table_add_row(node_table, flags, time, population, individual,
+                name, strlen(name));
+        assert(ret >= 0);
```

Comment:
Check and goto out. Nothing to be gained here from not doing it, as ``add_row`` will check its parameters and potentially return errors.